core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -715,6 +715,10 @@ class WorkspaceService(protected val ctx: RawlsRequestContext,
         } else Future.successful()
       )
 
+      _ <- traceWithParent("deleteWorkspaceInWSM", parentContext) { _ =>
+        maybeDeleteWsmWorkspace(workspaceContext)
```

Comment:
I've moved this up in the chain of calls. My reasoning is that we want to "unroll" the Azure workspace bottom up, so if the WSM record fails to delete, the Rawls record and Sam resource remain and we're able to retry.